web/src/stream_topic_history.ts

```diff
@@ -366,3 +366,26 @@ export function add_request_pending_for(stream_id: number): void {
 export function remove_request_pending_for(stream_id: number): void {
     request_pending_stream_ids.delete(stream_id);
 }
+
+// We only try to reify the message if it is still the last message
+// of a topic in the history.
```

Comment:
This should probably be rewritten to make clear why this is correct -- which is that this module only stores the last message ID.